symphony-bdk-core/src/main/java/com/symphony/bdk/core/auth/impl/AuthenticationRetry.java

```diff
@@ -82,7 +86,23 @@ public T executeAndRetry(String name, SupplierWithApiException<T> supplier, Stri
       }
       throw new ApiRuntimeException(e);
     } catch (Throwable t) {
-      throw new RuntimeException(t);
+      if (t.getCause() instanceof SocketTimeoutException || t.getCause() instanceof ConnectException) {
+        String timeoutMessageError = String.format("Failed while trying to connect to the \"%s\" at the following address: %s . "
+            + "Please check that the address is correct and make sure the service is up and running.", getServiceName(address), address);
+        throw new RuntimeException(timeoutMessageError, t);
+      } else {
+        throw new RuntimeException(t);
+      }
+    }
+  }
+
+  private String getServiceName(String address) {
```

Comment:
mm this looks fragile (and a bit of magic), is there a way we could get the service from the api client used? (ie because it is AuthenticationApi it is the KM, if it is PodApi it is the pod etc)

Reply:
yes this one I was planning to improve it, what I did here is to check the ApiClientFactory and see which are the existing ApiClient but for sure it needs adjustments, I'm working on it.

Reply:
@symphony-youri The problem here is that I'm not sure on how to get this information. The ApiClient does not contain the information on which component we are targeting. For example for the login we have this:
![Screenshot 2021-01-27 at 11 01 36](https://user-images.githubusercontent.com/65179248/105975309-1bec6880-608f-11eb-9ceb-c3a27aac0d27.png)
The only thing that I can think of is to add a new field and set it explicitly at ApiClient creation.